aws/subscriber.go

```diff
@@ -220,11 +230,7 @@ func (s *awsSubscriber) ensureFilterPolicy(filter map[string]string) error {
 		*/
 		if len(filter) == 0 {
 			log.Printf("AWS: clearing filter policy for topic %q, subID %q", s.topic, s.subscriptionID)
-			if _, err := s.sns.Unsubscribe(&sns.UnsubscribeInput{
-				SubscriptionArn: s.subscriptionArn,
-			}); err != nil {
-				return err
-			}
+			s.DeleteSubscription()
```

Comment:
good catch! But we shuld check for the `err` here instead of ignoring it

Reply:
Good call @dmacthedestroyer ! Fixed now!